### ISSUE TITLE
TEI writer extension (like element_citations for Jats)

### ISSUE THREAD

**Issue by marviro** [jgm/pandoc#8790]
The TEI writer does not keep the structured bibliographic information provided by citeproc.

This is possible using the JATS writer with the extension `element_citations`.

Would it be possible to create such an extension also for TEI?

The output, following the [TEI documentation](https://tei-c.org/release/doc/tei-p5-exemplars/html/tei_lite.doc.html#bibl), could be something like:

```
<bibl type="article" subtype="book_chapter"
 xml:id="carlin_2003">
 <author>
  <name>
   <surname>Carlin</surname>
     (<forename>Claire</forename>)</name>
 </author>,
<title level="a">The Staging of Impotence : France’s last
   congrès</title> dans
<bibl type="monogr">
  <title level="m">Theatrum mundi : studies in honor of Ronald W.
     Tobin</title>, éd.
 <editor>
   <name>
    <forename>Claire</forename>
    <surname>Carlin</surname>
   </name>
  </editor> et
 <editor>
   <name>
    <forename>Kathleen</forename>
    <surname>Wine</surname>
   </name>
  </editor>,
 <pubPlace>Charlottesville, Va.</pubPlace>,
 <publisher>Rookwood Press</publisher>,
 <date when="2003">2003</date>.
 </bibl>
</bibl>
```

Is the extension the best way to go, or would you rather suggest to write a filter? I guess that in this case it makes sense to have an extension for the writer...

**Comment by tarleb** [comment]
I agree that an extension makes sense. Could you link me to the relevant docs for this?

Related issue: jgm/citeproc#132; cc @frederik-elwert.

**Comment by marviro** [comment]
Thank you!
I think everything is in the TEI Lite doc, [here](https://tei-c.org/release/doc/tei-p5-exemplars/html/tei_lite.doc.html#bibl),

**Comment by tarleb** [comment]
I may have answered to quickly, because I didn't fully understand the difference between TEI and JATS. From what I gather, TEI citations are what's called "mixed citations" in JATS; there seems to be no equivalent to JATS element citations. That's problematic, because it means that we cannot just write the tags in any order, but that the CSL must always be applied. This isn't possible yet (and the subject of the citeproc issue linked above).

**Comment by marviro** [comment]
Ok, I understand. I am going to try to find a solution. If it is mandatory to apply a style, the other solution would be to create a filter which directly parses the structured bibliography, I guess? This is what you do with JATS ([here](https://github.com/jgm/pandoc/blob/main/src/Text/Pandoc/Writers/JATS.hs#L82))?

**Comment by YvesMarcoux** [comment]
@tarleb It's true that TEI's <bibl> are similar to JATS's "mixed citations". However, there is another TEI element <biblStruct> that is more structured and could likely be produced without applying the CSL. (<biblStruct> is not in TEI Lite, but it is in the "core" module of the TEI, so in theory in all flavours of the TEI).

Examples of this element are here: https://tei-c.org/release/doc/tei-p5-doc/en/html/examples-biblStruct.html

Does that look like a promising avenue?

**Comment by tarleb** [comment]
That's perfect. I'll give it a try when I have time.

**Comment by marviro** [comment]
I agree. But that would mean : 1) changing the CiteprocOutput class and 2) creating a new writer. I would definitely love that. (It's what we do with [Stylo](https://stylo.huma-num.fr/) an online text editor that uses pandoc (+some xslt) to produce output for journals (also TEI journal) and this is why we need a structured bibliography.)

**Comment by YvesMarcoux** [comment]
I believe a writer (or writer extension) is the best way to address that kind of needs in TEI outputs, but with a challenge. Essentially, there are just too many ways in which semantic bibliographic information can be encoded in the TEI (this is true with both `biblStruct`and `bibl`). Thus, a writer/extension is doomed to be useful (at least directly) only in those use cases (i.e. TEI projects) that happen to use the same representation as the one used by the writer/extension. Please see https://github.com/jgm/citeproc/issues/132#issuecomment-3391770947 for the full argument.

In our current (MD, BibLaTeX) -> TEI project, instead of engaging in writing a writer/extension (or waiting for someone to do it), we are considering processing the bibliography and the MD in parallel:

  (MD, BibLaTeX) -> (pandoc + citeproc) -> XML
  BibLaTeX -> pandoc -> CLSJSON

Then, use XSLT to reconcile everything in our own little flavor of TEI. citeproc will be used to manage citations and cites.

I cannot resist the temptation of saying that this "own little flavor" has some normative potential, as the project involves north-american and european publication platforms.
